driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -251,36 +372,49 @@ public void delete(T entity) {
     }
 
     /**
-     * Deletes an entity based on its primary key.
+     * Deletes an entity mapped by this mapper asynchronously.
      * <p>
-     * This method is basically equivalent to: {@code getManager().getSession().execute(deleteQuery(primaryKey))}.
+     * This method is basically equivalent to: {@code getManager().getSession().executeAsync(deleteQuery(entity, options))}.
      *
-     * @param primaryKey the primary key of the entity to delete, or more precisely
-     * the values for the columns of said primary key in the order of the primary key.
+     * @param entity the entity to delete.
+     * @return a future on the completion of the deletion.
+     */
+    public ListenableFuture<Void> deleteAsync(T entity, Option... options) {
+        return Futures.transform(session().executeAsync(deleteQuery(entity, options)), NOOP);
+    }
+
+    /**
+     * Deletes an entity based on its primary key.
+     * <p>
+     * This method is basically equivalent to: {@code getManager().getSession().execute(deleteQuery(objects))}.
      *
+     * @param objects the primary key of the entity to delete, or more precisely
+     *                the values for the columns of said primary key in the order
+     *                of the primary key.Can be followed by {@link Option} to include
+     *                in the DELETE query.
      * @throws IllegalArgumentException if the number of value provided differ from
-     * the number of columns composing the PRIMARY KEY of the mapped class, or if
-     * at least one of those values is {@code null}.
+     *                                  the number of columns composing the PRIMARY KEY of the mapped class, or if
+     *                                  at least one of those values is {@code null}.
      */
-    public void delete(Object... primaryKey) {
-        session().execute(deleteQuery(primaryKey));
+    public void delete(Object... objects) {
```

Comment:
It annoys me that we need this varargs containing different types of object, but I can't think of anything better at the moment.